packages/ra-core/src/dataProvider/useCreate.ts

```diff
@@ -91,7 +91,7 @@ export const useCreate = <
             meta: callTimeMeta = paramsRef.current.meta,
         } = {}) =>
             dataProvider
-                .create<RecordType>(callTimeResource, {
+                .create<RecordType>(callTimeResource as string, {
```

Comment:
This is a bit of a hack. I assume the intention is that you either have to provide `resource` in `useCreate()` invocation, or you need to pass it to returned mutation function:

1)
```javascript
const [create] = useCreate('product')
const doCreateProduct = useCallback((data) => create({data}));
```

2)
```javascript
const [create] = useCreate();
const doCreateProduct = useCallback((data) => create({resource: 'product', data}));
```

Only these 2 cases should be allowed. If that's the intention, then I see 2 options:
1) Make `UseCreateResult` result conditional based on the provided `resource`, so that it's required param in the returned mutation function if it's `undefined`, or optional if it's not `undefined`
2) Just do a runtime check and don't call `dataProvider` if `callTimeResource` is undefined.

WDYT?